src/hardware/setup.rs

```diff
@@ -313,6 +316,54 @@ pub fn setup(
         }
     };
 
+    info!("setup internal ADCs");
+
+    let mut adc_int = AdcInternal::new(
+        &mut delay,
+        &ccdr.clocks,
+        ccdr.peripheral.ADC12,
+        ccdr.peripheral.ADC3,
+        device.ADC1,
+        device.ADC2,
+        device.ADC3,
+        gpioc.pc0.into_analog(),
+        gpioc.pc2.into_analog(),
+        gpiof.pf7.into_analog(),
+        gpiof.pf8.into_analog(),
+        gpioc.pc3.into_analog(), // untested on HW
+        gpioa.pa0.into_analog(), // untested on HW
+        gpioa.pa3.into_analog(), // untested on HW
+        gpioa.pa4.into_analog(), // untested on HW
+        gpioa.pa5.into_analog(), // untested on HW
+        gpioa.pa6.into_analog(), // untested on HW
+        gpiob.pb0.into_analog(), // untested on HW
+        gpiob.pb1.into_analog(), // untested on HW
+    );
+
+    info!(
+        "5V: {:?}V",
+        (adc_int.read(IAdc::Supply(Supply::P5v)) as f32 / (1 << 16) as f32) as f32
```

Comment:
All these should look like `read() as f32 * (thing_per_lsb)`. With a `let thing_per_lsb = ...` before that.
And methods to do that conversion plus the constants should go into `adc_int`.

Reply:
Also stick to the example with `max_sample()`. https://github.com/stm32-rs/stm32h7xx-hal/blob/5e35866c6e95a23eeafc3b5a6ab70a9f1bfd9e7e/examples/adc.rs#L79

Reply:
for now in individual functions